clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -366,12 +366,12 @@ public String transactionalId() {
         return transactionalId;
     }
 
-    public boolean isTransactional() {
```

Comment:
fwiw, the renamed methods are a breaking change of public API and shouldn't have been merged into master as is, I think. Why not just `@deprecate` them? Asking, because this has caused me some headache today :(

Reply:
The request classes are not public API. Only the classes shown in the javadoc are considered public:
https://kafka.apache.org/21/javadoc/overview-summary.html

Out of curiosity, how are you using this internal classes?

Reply:
Thanks for getting back to me. We're using [Alpakka Kafka](https://doc.akka.io/docs/akka-stream-kafka/1.0-RC2/) and I wanted to test the client-version 2.1.1 talking with server 2.0.1. 

For testing purposes we're spinning up an embedded Kafka, which comes from a transitive dependency that [`akka-stream-kafka-testkit`](https://doc.akka.io/docs/akka-stream-kafka/1.0-RC2/testing.html) provides. These APIs are used somewhere in there, so it's not even code that I control.

I wasn't aware that this is considered "private public" API and was a bit worried about the "unnoticed" breaking changes to the API. Thanks for the explanation